rust/src/rune_coral.rs

```diff
@@ -103,14 +83,4 @@ fn dummy_tensors(inputs: &[TensorDescriptor<'_>]) -> Vec<ffi::RuneCoralTensor> {
     }
 
     tensors
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn library_wrapper_is_send_and_sync() {
-        static_assertions::assert_impl_all!(RuneCoral: Send, Sync);
```

Comment:
I know we don't need this test any more because `RuneCoral` has been removed, but could we add an equivalent one for the `InferenceContext` type?

I believe we want to assert that `InferenceContext: Send` because it doesn't have any thread-local state or other reason to be pinned to a thread, but we need to make sure it is `!Sync` because it doesn't do any internal synchronisation (i.e. not thread-safe).

That should be fine for Rune's needs, although it'll need to wrap an `InferenceContext` in a mutex to prevent concurrent access when evaluating the WebAssembly code.

Reply:
Will implement